clcache.py

```diff
@@ -817,29 +817,27 @@ def analyze(cmdline):
                 return AnalysisResult.MultipleSourceFilesComplex, None, None
             return AnalysisResult.MultipleSourceFilesSimple, sourceFiles, None
 
-        outputFile = None
-        if 'Fo' in options:
-            outputFile = options['Fo'][0]
-            outputFile = os.path.normpath(outputFile)
-
-            if os.path.isdir(outputFile):
-                outputFile = os.path.join(outputFile, basenameWithoutExtension(sourceFiles[0]) + ".obj")
-
-        elif preprocessing:
-            if 'P' in options:
-                # Preprocess to file.
+        if preprocessing:
+            if 'E' in options or 'EP' in options:
+                # Preprocess to stdout
+                outputFile = None
+            else:
+                # Preprocess to file
                 if 'Fi' in options:
                     outputFile = options['Fi'][0]
                 else:
                     outputFile = basenameWithoutExtension(sourceFiles[0]) + ".i"
-            else:
-                # Preprocess to stdout. Use empty string rather then None to ease
-                # output to log.
-                outputFile = ''
```

Comment:
Ah thanks for looking at the source. [When vchigrin commited this](https://github.com/frerich/clcache/blob/a37710335e62e98a3140757743fa161f2d818057/clcache.py), we introduced the first non-path outputFile value and the rest of the code was not ready to deal with it. [Existing](https://github.com/frerich/clcache/blob/a37710335e62e98a3140757743fa161f2d818057/clcache.py#L777) [tracing](https://github.com/frerich/clcache/blob/a37710335e62e98a3140757743fa161f2d818057/clcache.py#L791) and [os path](https://github.com/frerich/clcache/blob/a37710335e62e98a3140757743fa161f2d818057/clcache.py#L779) [methods](https://github.com/frerich/clcache/blob/a37710335e62e98a3140757743fa161f2d818057/clcache.py#L790) could not deal with `None`. So he did a plausible thing in order to get a minimal commit.